### ISSUE TITLE
[BottomSheetDialogFragment/NestedScrollView] Weird behavior when tapping on WebView inside NestedScrollView inside BottomSheetDialogFragment

### ISSUE THREAD

**Issue by tfcporciuncula** [material-components/material-components-android#2219]
**Description:**

We have a `WebView` in a `BottomSheetDialogFragment`, and my understanding is that it’s a good idea to have it inside a `NestedScrollView` so when the bottom sheet is expanded and you scroll up in the `WebView`, you actually scroll the `WebView` content instead of dragging the bottom sheet down.

Everything works great except for this very specific thing: our `WebView` has some top margins (so the rounded corners look good), and when you tap on the `WebView`, the content immediately goes up as if there’s no top margin. It’s easier to visualize the issue in a video:

https://user-images.githubusercontent.com/750114/119615953-53321280-be00-11eb-826a-9bcb20084186.mov

So when we tap on the `WebView` (any part of its content) for the first time, we basically go from here to here:

![a](https://user-images.githubusercontent.com/750114/119616109-8674a180-be00-11eb-8c03-ec442117c831.png) ![b](https://user-images.githubusercontent.com/750114/119616114-870d3800-be00-11eb-98ce-9beaf3f1ca8f.png)

And it’s like the content just scrolled for some reason, because you can scroll a bit and reach this weird state here:

![c](https://user-images.githubusercontent.com/750114/119616233-a86e2400-be00-11eb-9802-3781e48bbee9.png)

The amount it scrolls always matches exactly the margin top I have in the `WebView`, and it’s always the same result regardless of how I define this space (top margin on `WebView`, padding top on `WebView` parent, `Space` widget, etc).

If instead of a `WebView` I have other kind of scrollable content, the issue doesn't seem to happen, so the issue really seems to be related to the `WebView` somehow.

**Expected behavior:** Nothing happens when I simply tap on the `WebView`.

**Source code:** https://github.com/tfcporciuncula/playground/tree/bottomsheet-issue (`bottomsheet-issue` branch)

**Android API version:** 29

**Material Library version:** 1.3.0 and 1.4.0-beta01

**Device:** Pixel 3, seems to happen consistently on any device

**Comment by drchen** [comment]
If you add a Space on top of WebView instead of using marginTop, will it solve the issue?

**Comment by tfcporciuncula** [comment]
Nope. As I mentioned above, it doesn't matter how I define the space at the top (margin, padding, `Space` widget) the issue persists.

**Comment by marlonlml1308** [comment]
link source code, not found
